pb/protocol_messenger.go

```diff
@@ -170,7 +169,7 @@ func (pm *ProtocolMessenger) GetClosestPeers(ctx context.Context, p peer.ID, id
 }
 
 // PutProvider asks a peer to store that we are a provider for the given key.
-func (pm *ProtocolMessenger) PutProvider(ctx context.Context, p peer.ID, key multihash.Multihash, host host.Host) (err error) {
+func (pm *ProtocolMessenger) PutProvider(ctx context.Context, p peer.ID, key multihash.Multihash, self peer.AddrInfo) (err error) {
```

Comment:
I'm using the protocol messenger directly in a couple of tools (e.g., Nebula). I'm not using this API specifically but I wouldn't assume others also don't use it. This change would break them.

Reply:
Do you want me to make an other method ?

Reply:
I think your signature makes much more sense instead of passing the whole host into that method. Just wanted to make sure we're aware of the consequences.

Reply:
Yeah I just assumed no one would depend on this code, I'm good making this change backward compatible if you want.

Reply:
Is there a way to know how many projects depend on this method? If many projects are using it, I am in favor of making the change backward compatible 👍🏻

Reply:
There is https://pkg.go.dev/github.com/libp2p/go-libp2p-kad-dht/pb?tab=importedby which suggest 68, but this is at a package level.

Reply:
Then let's not break the users :) I am good to merge after this change becomes non-breaking.

Reply:
Done